pkg/controller/remotemachineset/gcpactuator.go

```diff
@@ -244,6 +242,41 @@ func (a *GCPActuator) obtainLease(pool *hivev1.MachinePool, cd *hivev1.ClusterDe
 		if err != nil {
 			return "", false, err
 		}
+		if len(availLeaseChars) == 0 {
+			logger.Warn("no GCP MachinePoolNameLease characters available, setting condition")
+			conds, changed := controllerutils.SetMachinePoolConditionWithChangeCheck(
+				pool.Status.Conditions,
+				hivev1.NoMachinePoolNameLeasesAvailable,
+				corev1.ConditionTrue,
+				"AllCharactersInUse",
```

Comment:
I would rather the reason and message not mention characters and instead say out of machine pool names. That way, if we end up having to use this for some other platform that may have more than 1 character, we can still use the same reason and message.